include/stl2/detail/iterator/ostream_iterator.hpp

```diff
@@ -26,63 +26,58 @@
 #include <stl2/detail/memory/addressof.hpp>
 
 STL2_OPEN_NAMESPACE {
-	namespace detail {
-		///////////////////////////////////////////////////////////////////////////
-		// ostream_cursor [Implementation detail]
-		//
-		template <class T, class charT, class traits>
-		requires
-			models::Same<T, void> ||
-			models::StreamInsertable<T, std::basic_ostream<charT, traits>>
-		class ostream_cursor {
+	///////////////////////////////////////////////////////////////////////////
+	// ostream_iterator [ostream.iterator]
+	// Extension: ostream_iterator<void> accepts any streamable type.
+	// Not to spec:
+	// * StreamInsertable constraint is implicit
+	//   See https://github.com/ericniebler/stl2/issues/246
+	//
+	template <class T = void, class charT = char, class traits = std::char_traits<charT>>
+	requires
+		models::Same<T, void> ||
+		models::StreamInsertable<T, std::basic_ostream<charT, traits>>
+	class ostream_iterator {
 		public:
-			using difference_type = std::ptrdiff_t;
-			using ostream_type = std::basic_ostream<charT, traits>;
-			struct mixin : protected ebo_box<ostream_cursor> {
-				using difference_type = ostream_cursor::difference_type;
-				using char_type = charT;
-				using traits_type = traits;
-				using ostream_type = ostream_cursor::ostream_type;
-				using ebo_box<ostream_cursor>::ebo_box;
-			};
+			typedef ptrdiff_t difference_type;
+			typedef charT char_type;
+			typedef traits traits_type;
+			typedef std::basic_ostream<charT, traits> ostream_type;
+
+			ostream_iterator() = default;
 
-			constexpr ostream_cursor() noexcept = default;
-			STL2_CONSTEXPR_EXT ostream_cursor(
-				ostream_type& os, const char* delimiter = nullptr)
-			noexcept
-			: os_{__stl2::addressof(os)}, delimiter_{delimiter}
-			{}
+			STL2_CONSTEXPR_EXT ostream_iterator(
+				ostream_type& os, const charT* delimiter = nullptr) noexcept
+			: out_stream(__stl2::addressof(os)), delim(delimiter) {}
 
 			template <class U, class V = meta::if_<std::is_void<T>, U, T>>
 			requires
 				ConvertibleTo<U, V const&>() &&
 				ext::StreamInsertable<V, ostream_type>
-			STL2_CONSTEXPR_EXT void write(U&& u) {
-				*os_ << static_cast<V const &>(std::forward<U>(u));
-				if (delimiter_) {
-					*os_ << delimiter_;
+			ostream_iterator& operator=(U&& u) {
+				*out_stream << static_cast<V const &>(std::forward<U>(u));
+				if (delim != nullptr) {
+					*out_stream << delim;
 				}
+				return *this;
 			}
 
+			ostream_iterator& operator*() {
+				return *this;
+			}
+			ostream_iterator& operator++() {
+				return *this;
+			}
+			// Not to spec:
+			// https://github.com/ericniebler/stl2/issues/232
+			ostream_iterator& operator++(int) {
+				return *this;
+			}
 		private:
-			raw_ptr<ostream_type> os_ = nullptr;
-			const char* delimiter_ = nullptr;
-		};
-	}
+			detail::raw_ptr<basic_ostream<charT, traits>> out_stream{nullptr};
+			const charT* delim{nullptr};
```

Comment:
non-static data members have names that end with `_`. Yes, I know these are the exposition-only names given to these members in the TS. No, I don't think it will cause confusion to name them `out_stream_` and `delim_`.